### PR TITLE
Add a bucket and mop slot to the Janicart

### PR DESCRIPTION
**Pull request by chrome-cirrus**
## About the PR
-Adds a bucket and mop slot to the Janicart, upgrading the arsenal of our valiant janitorial warriors to better combat the messy nature of the vastness of space.

## Why / Balance
No bigtime balance issues. Came out of some idle chatter on Discord.

https://discord.com/channels/1123826877245694004/1123851369913266256/1314977814918860810
https://discord.com/channels/1123826877245694004/1127687656084611214/1315002359511781476

## How to test
Spawn a Janicart NF variant in, check functions. Check out mapped locations of the cart to see the NF version is there. Check out dungeon spawners and Janicart crates and see they spawn the NF version.

## Media
crate
examine
mopeject
mopinsert
moptry

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes
Nothing really

**Changelog**
:cl:
- tweak: Add slots to the Janicart vehicle to give it feature parity with the janitor trolley and make things more convenient for practicioners of the janitorial arts
- add: Research target that unlocks crafting of Janicart flatpacks
- remove: Janicart crates no longer purchaseable at cargo

### CODE REVIEW

**Comment by chrome-cirrus** [comment]
Setting to draft for the moment, I want to also make the Janicart researchable as part of this.

**Comment by chrome-cirrus** [comment]
There we go. Added a version that comes with its key in the ignition, a flatpack that creates one, and a research path to unlock production of them. 

Functionality-wise, I think this is all good to go.

Aesthetically, sprite-enhancement to add item mapping for mops onto the Janicart sprite and/or somehow showing a visual indicator of its bucket fullness would be neat, but I don't think it's a showstopper to not have those.

Cash values, material requirements, and research point costs are 100% vibes based and I don't have strong opinions on their correctness. Would appreciate feedback there.

flatpack
keyedvariant
lathe
research

File: Resources/Locale/en-US/_NF/research/technologies.ftl [hunk 1]
```diff
@@ -13,4 +13,5 @@ research-technology-bounty-hunting = Bounty Hunting
 research-technology-arsenal-style = Punk Gear
 research-technology-industrial-medicine = Industrial Medicine
 research-technology-magnets-tech-advanced = Advanced Localized Magnetism
-research-technology-magnets-tech-combat = Localized Magnetism Combat Application
\ No newline at end of file
+research-technology-magnets-tech-combat = Localized Magnetism Combat Application
+research-technology-janicart-fabrication = Janicart Fabrication
```
Comment (whatston3) [inline Resources/Locale/en-US/_NF/research/technologies.ftl:17]: Mobile Sanitation?

File: Resources/Prototypes/Entities/Structures/Machines/lathe.yml [hunk 2]
```diff
@@ -406,6 +406,7 @@
       - ConstructionBagOfHolding # Frontier
       - ServiceSelectiveDropper # Frontier
       - PlantAnalyzer # Frontier
+      - JanicartFlatpack # Frontier
```
Comment (whatston3) [inline Resources/Prototypes/Entities/Structures/Machines/lathe.yml:409]: Could throw it in the service techfab for giggles.

File: Resources/Prototypes/_NF/Entities/Objects/Vehicles/vehicles.yml [hunk 3]
```diff
@@ -183,6 +183,166 @@
           params:
             volume: -3
 
+# Enhanced version of the vanilla Janicart that adds a bucket and mop slot.
+- type: entity
+  id: VehicleJanicartNF
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/Objects/Vehicles/vehicles.yml:188]: Prefer prefixes to suffixes.  I know we've got both, but prefer prefixes to suffixes.

File: Resources/Prototypes/_NF/Entities/Objects/Vehicles/vehicles.yml [hunk 4]
```diff
@@ -183,6 +183,166 @@
...
+        insertSound:
+          path: /Audio/Effects/Vehicle/vehiclestartup.ogg
+          params:
+            volume: -3
+      trashbag_slot:
+        name: janitorial-trolley-slot-component-slot-name-trashbag
+        whitelist:
+          tags:
+            - TrashBag
+        priority: 3 # Higher than drinking priority
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/Objects/Vehicles/vehicles.yml:268]: Why not go for parity with the trolley itself, add slots for wet floor signs, anything else it has?  Hell, you might be able to reparent from both the Janicart and the janitorial trolley without much issue.

**Review by whatston3 — COMMENTED** [review]
Idea seems fine, why not take more things from the trolley if you're trying to be an upgraded version of it?

Also, janicart crates can still be purchased from cargo, right?  Are you removing that and making it an upgradeable purchase?  If so, abstract the cargo listing - makes sense to me, though.  I'd double check for references to that crate if they exist, unsure about it off the top of my head.

File: Resources/Locale/en-US/_NF/research/technologies.ftl (hunk 1 above)
Comment (chrome-cirrus) [reply Resources/Locale/en-US/_NF/research/technologies.ftl:17]: I like that! Will steal.

File: Resources/Prototypes/Entities/Structures/Machines/lathe.yml (hunk 2 above)
Comment (chrome-cirrus) [reply Resources/Prototypes/Entities/Structures/Machines/lathe.yml:409]: Agreed, gonna do it.

File: Resources/Prototypes/_NF/Entities/Objects/Vehicles/vehicles.yml (hunk 3 above)
Comment (chrome-cirrus) [reply Resources/Prototypes/_NF/Entities/Objects/Vehicles/vehicles.yml:188]: Got it, will adjust that!

File: Resources/Prototypes/_NF/Entities/Objects/Vehicles/vehicles.yml (hunk 4 above)
Comment (chrome-cirrus) [reply Resources/Prototypes/_NF/Entities/Objects/Vehicles/vehicles.yml:268]: Felt a bit weird to have _that_ much stuff tacked onto it given how sleek the vehicle design is... But eh why not. 

I'll have to try out the multi-parenting. Would be slick if it Just Worked(tm).

**Comment by dvir001** [comment]
@chrome-cirrus NF always a prefix now, not a suffix, please fix it for all your added entity.

**Comment by chrome-cirrus** [comment]
> @chrome-cirrus NF always a prefix now, not a suffix, please fix it for all your added entity.
> 
> Eg, NFVehicleJanicart

It is done.

**Comment by chrome-cirrus** [comment]
> Idea seems fine, why not take more things from the trolley if you're trying to be an upgraded version of it?
> 
> Also, janicart crates can still be purchased from cargo, right? Are you removing that and making it an upgradeable purchase? If so, abstract the cargo listing - makes sense to me, though. I'd double check for references to that crate if they exist, unsure about it off the top of my head.

We have slot parity!

moreslots

Only reference to the crate I found was the cargo menu item. Removed that, with them becoming craftable seems to make sense (and there are still plenty of them on expeds and POIs too).

File: Resources/Prototypes/_NF/Entities/Objects/Vehicles/vehicles.yml (hunk 4 above)
Comment (chrome-cirrus) [reply Resources/Prototypes/_NF/Entities/Objects/Vehicles/vehicles.yml:268]: Multi-parenting didn't work smoothly for me, but I added the additional slots.

**Comment by whatston3** [comment]
I put up a set of suggested changes on [whatston3:2024-12-17-janicart-suggestions](https://github.com/chrome-cirrus/frontier-station-14/compare/janicart...whatston3:frontier-station-14:2024-12-17-janicart-suggestions?expand=1)

In summary:
* Slots defined using YAML anchors where possible to cut down on lines of definition
* Base janicart abstracted
* Water removed from Janicart
* NFJanicart inherits from base one
* Migration moved to _NF
* Key priority changed to 10 vs. 9 (just to be higher than the mop)

Generally seems fine to me, though the value of the cart seems high.

**Comment by chrome-cirrus** [comment]
Thanks @whatston3! Great suggestions, pulled all of that in. I can't believe I didn't think about anchors, considering how much I use yaml day to day. That saves a ton of lines.

Value was indeed way high. I took it way down, flatpack is now worth 1.25x the raw materials that go into crafting it, and the deployed cart 1.2x the raw material cost for the flatpack.

**Comment by whatston3** [comment]
I'd leave the values 1:1 between the flatpack and the machine, 375 seems fine.
Didn't see the guidebook reference for "VehicleJanicart", my mistake.

**Comment by chrome-cirrus** [comment]
Made the guidebook reference the NF prefixed version. Whatever parses the guidebook XML for rendering does not care for comment tags so, not sure what the preferred way to note NF specific stuff in there is.

**Review by whatston3 — APPROVED** [review]
Looks good to me.

**Pull request merged**